stubs/docker/docker/client.pyi

```diff
@@ -12,11 +13,25 @@ from docker.models.services import ServiceCollection
 from docker.models.swarm import Swarm
 from docker.models.volumes import VolumeCollection
 
+@type_check_only
+class _Environ(Protocol):
+    def __getitem__(self, k: str, /) -> str: ...
+    def keys(self) -> Iterable[str]: ...
+
+@type_check_only
+class _FromEnvDict(TypedDict):
+    version: NotRequired[str]
+    timeout: NotRequired[int]
+    max_pool_size: NotRequired[int]
+    environment: NotRequired[_Environ]
```

Comment:
I chose to create / use the `_Environ` protocol here.
The documentation says:

```python
            environment (dict): The environment to read environment variables
                from. Default: the value of ``os.environ``
```

And the implementation includes:

```
    if not environment:
        environment = os.environ
```

I therefore considered `dict[str, str]`, but I thought it best to allow users to pass in `os.environ`.